Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/MaterialComponentController.cpp

```diff
@@ -111,6 +114,55 @@ namespace AZ
         {
             InitializeMaterialInstance(asset);
         }
+        
+        void MaterialComponentController::OnModelReady(const Data::Asset<RPI::ModelAsset>&, const Data::Instance<RPI::Model>&)
+        {
+            MeshComponentNotificationBus::Handler::BusDisconnect();
+
+            // If there is a circumstance where the saved material assignments are empty, fill them in with the default material.
+            // (This could happen as a result of LoadMaterials() clearing the asset reference to deal with an edge case)
+
+            // Now that a model asset is ready, see if there are any empty assignments that need to be filled...
+            RPI::ModelMaterialSlotMap modelMaterialSlots;
+            MaterialReceiverRequestBus::EventResult(modelMaterialSlots, m_entityId, &MaterialReceiverRequestBus::Events::GetModelMaterialSlots);
+
+            AZStd::vector<Data::Asset<RPI::MaterialAsset>> newMaterialAssets;
+            newMaterialAssets.reserve(m_configuration.m_materials.size());
+
+            // First we fill the empty slots but don't connect to AssetBus yet. If the same material asset appears multiple times,
+            // AssetBus will call OnAssetReady only the *first* time we connect for that asset. The full list of m_configuration.m_materials
+            // needs to be updated before that happens.
+            for (auto& materialPair : m_configuration.m_materials)
+            {
+                auto& materialAsset = materialPair.second.m_materialAsset;
+
+                if (!materialAsset.GetId().IsValid())
+                {
+                    auto slotIter = modelMaterialSlots.find(materialPair.first.m_materialSlotStableId);
+                    if (slotIter != modelMaterialSlots.end())
+                    {
+                        materialAsset = slotIter->second.m_defaultMaterialAsset;
+                        newMaterialAssets.push_back(materialAsset);
+                    }
+                    else
+                    {
+                        AZ_Error("MaterialComponentController", false, "Could not find material slot %d", materialPair.first.m_materialSlotStableId);
+                    }
+                }
+            }
+
+            // Now that the configuration is updated with all the default material assets, we can load and connect them.
+            // If there are duplicates in this list, the redundant calls will be ignored.
+            for (auto& materialAsset : newMaterialAssets)
+            {
+                if (!materialAsset.IsReady())
+                {
+                    materialAsset.QueueLoad();
+                }
+
+                Data::AssetBus::MultiHandler::BusConnect(materialAsset.GetId());
+            }
+        }
```

Comment:
First, see my comments below.
Regarding "Material component also cannot have a direct dependency on the mesh component", I am not making a hard dependency here. If the Material Component is used with a component that provides a ModelAsset, then it can use this signal. (any component that provides a ModelAsset should be sending notifications via MeshComponentBus, though maybe it should be renamed as I mentioned above).